File: 10º Semestre/Sistemas Distribuidos/Exp4/exercicio 5/app/python/main.py
import argparse
import person_pb2
from google.protobuf.json_format import MessageToJson, Parse

def encode():
    p = person_pb2.Person()
    p.Name = "Angelo Duarte"
    p.EnrollNumber = 123456
    p.Height = 1.77
    p.LuckNumbers.extend([7, 23, 47])
    j = MessageToJson(p, including_default_value_fields=False, preserving_proto_field_name=False)
    print(j)

def decode(data):
    p = person_pb2.Person()
    Parse(data, p)
    print(f'Name: {p.Name}')
    print(f'EnrollNumber: {p.EnrollNumber}')
    print(f'Height: {p.Height}')
    print(f'LuckNumbers: {p.LuckNumbers}')

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('args', nargs=argparse.REMAINDER)
    args = parser.parse_args()

    if args.args: decode(args.args[0])
    else: encode()
        

File: 10º Semestre/Sistemas Distribuidos/Exp4/exercicio 5/app/java/Main.java
import demo.PersonProto.Person;
import com.google.protobuf.util.JsonFormat;
import com.google.protobuf.InvalidProtocolBufferException;

import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        if (args.length == 0) encode();
        else decode(args[0]);
    }

    private static void encode() throws Exception {
        Person p = Person.newBuilder()
                .setName("Angelo Duarte")
                .setEnrollNumber(123456)
                .setHeight(1.77)
                .addAllLuckNumbers(Arrays.asList(7, 23, 47))
                .build();

        String json = JsonFormat.printer().print(p);
        System.out.println(json);
    }

    private static void decode(String json) throws Exception {
        Person.Builder b = Person.newBuilder();
        JsonFormat.parser().ignoringUnknownFields().merge(json, b);
        Person p = b.build();

        System.out.println("Name: " + p.getName());
        System.out.println("EnrollNumber: " + p.getEnrollNumber());
        System.out.println("Height: " + p.getHeight());
        System.out.println("LuckNumbers: " + p.getLuckNumbersList());
    }
}

